apps/shinkai-desktop/src/components/agent/agent-form.tsx

```diff
@@ -317,28 +365,155 @@ function AgentSideChat({
     });
   };
 
+  const { data: agentInboxes } = useGetAgentInboxes({
+    agentId,
+    nodeAddress: auth?.node_address ?? '',
+    token: auth?.api_v2_key ?? '',
+  });
+
+  // State for the currently selected value in the dropdown
+  const [selectedInboxId, setSelectedInboxId] = useState<string>('new_chat');
+
+  // Effect to sync selectedInboxId with chatInboxId
+  useEffect(() => {
+    if (selectedInboxId === 'new_chat') {
+      setChatInboxId(null);
+    } else {
+      setChatInboxId(selectedInboxId);
+    }
+  }, [selectedInboxId]);
+
+  // Ref to track the previous agentInboxes value
+  const prevAgentInboxesRef = useRef(agentInboxes);
+
+  // Reset selectedInboxId only if the list genuinely changes and the selection is gone
+  useEffect(() => {
+    const hasInboxListChanged = prevAgentInboxesRef.current !== agentInboxes;
+
+    if (
+      hasInboxListChanged && // Only run if the list object reference changed
+      agentInboxes &&
+      selectedInboxId !== 'new_chat' &&
+      !agentInboxes.find((inbox) => inbox.inbox_id === selectedInboxId)
+    ) {
+      setSelectedInboxId('new_chat'); // Reset to new chat state
+    }
+
+    // Update the ref for the next render
+    prevAgentInboxesRef.current = agentInboxes;
+  }, [agentInboxes, selectedInboxId]);
+
+  // Effect to sync the dropdown selection AFTER the inbox list is updated
+  useEffect(() => {
+    if (chatInboxId && agentInboxes?.length) {
+      const activeChatExistsInList = agentInboxes.some(
+        (inbox) => inbox.inbox_id === chatInboxId,
+      );
+
+      if (activeChatExistsInList) setSelectedInboxId(chatInboxId);
+      else {
+        console.warn(
+          `Chat inbox ${chatInboxId} not found in agent inboxes`,
+          agentInboxes,
+        );
+      }
+    }
+  }, [agentInboxes, chatInboxId]); // Depend on both the list and the active chat ID
+
```

Comment:
I'd recommend reusing the existing chatInboxId state for the history feature. Some of the useEffect hooks here might be unnecessary and could be adding complexity that we can avoid

Reply:
Done